superset/db_engine_specs/starrocks.py

```diff
@@ -0,0 +1,236 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import logging
+import re
+from typing import Any, Dict, List, Optional, Pattern, Tuple, Type
+from urllib import parse
+
+from flask_babel import gettext as __
+from sqlalchemy import Integer, Numeric, types
+from sqlalchemy.engine import Inspector
+from sqlalchemy.engine.result import Row as ResultRow
+from sqlalchemy.engine.url import URL
+from sqlalchemy.sql.type_api import TypeEngine
+
+from superset.db_engine_specs import BaseEngineSpec
+from superset.db_engine_specs.mysql import MySQLEngineSpec
+from superset.errors import SupersetErrorType
+from superset.utils.core import GenericDataType
+
+# Regular expressions to catch custom errors
+CONNECTION_ACCESS_DENIED_REGEX = re.compile(
+    "Access denied for user '(?P<username>.*?)'"
+)
+CONNECTION_UNKNOWN_DATABASE_REGEX = re.compile("Unknown database '(?P<database>.*?)'")
+
+logger = logging.getLogger(__name__)
+
+
+class TINYINT(Integer):
+    __visit_name__ = "TINYINT"
+
+
+class DOUBLE(Numeric):
+    __visit_name__ = "DOUBLE"
+
+
+class ARRAY(TypeEngine):
+    __visit_name__ = "ARRAY"
+
+    @property
+    def python_type(self) -> Optional[Type[List[Any]]]:
+        return list
+
+
+class MAP(TypeEngine):
+    __visit_name__ = "MAP"
+
+    @property
+    def python_type(self) -> Optional[Type[Dict[Any, Any]]]:
+        return dict
+
+
+class STRUCT(TypeEngine):
+    __visit_name__ = "STRUCT"
+
+    @property
+    def python_type(self) -> Optional[Type[Any]]:
+        return None
+
+
+class StarRocksEngineSpec(MySQLEngineSpec):
+    engine = "starrocks"
+    engine_name = "StarRocks"
+
+    default_driver = "starrocks"
+    sqlalchemy_uri_placeholder = (
+        "starrocks://user:password@host:port/catalog.db[?key=value&key=value...]"
+    )
+
+    super().column_type_mappings = (
```

Comment:
```suggestion
    column_type_mappings = (
```
I don't think you can have `super()` here, check how the other db engine specs do it. The way this works, is `column_type_mappings` is checked first, and if there are no hits, it proceeds to `_default_column_type_mappings`.

Reply:
Yes, actually I didn't write `super()` originally, but mypy reports `superset/db_engine_specs/starrocks.py:71: error: Incompatible types in assignment (expression has type Tuple[Tuple[Pattern[str], TINYINT, GenericDataType], Tuple[Pattern[str], Any, GenericDataType], ... <14 more items>], base class "MySQLEngineSpec" defined the type as Tuple[Tuple[Pattern[str], Any, GenericDataType], Tuple[Pattern[str], Any, GenericDataType], Tuple[Pattern[str], Any, GenericDataType], Tuple[Pattern[str], Any, GenericDataType], Tuple[Pattern[str], Any, GenericDataType], Tuple[Pattern[str], Any, GenericDataType], Tuple[Pattern[str], Any, GenericDataType], Tuple[Pattern[str], Any, GenericDataType], Tuple[Pattern[str], Any, GenericDataType], Tuple[Pattern[str], Any, GenericDataType]])  [assignment]` 

I don't know how to fix this check, can you give some suggestion？